bus-mapping/src/circuit_input_builder/execution.rs

```diff
@@ -336,6 +338,34 @@ pub enum NumberOrHash {
     Hash(H256),
 }
 
+/// Represents all bytes related in one copy event.
+#[derive(Clone, Debug)]
+pub struct CopyBytes {
+    /// Represents the list of (bytes, is_code, mask) copied during this copy event
+    pub bytes: Vec<(u8, bool, bool)>,
```

Comment:
Since there is already a `value` in each copy step, that is a good place for `value_prev` too. In the same tuple, like this: 

`(value, value_prev, is_code, mask)`

And there is only `read_steps` and `write_steps`.

Then this `CopyBytes` structure is simpler or not needed. And all the changes below with the tuples `(copy_steps, prev_bytes)` can be avoided.

Reply:
already added in value_prev in CopyStep

Reply:
Correct, I meant that this same structure could be instead of `(u8, bool, bool)`